src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php

```diff
@@ -906,6 +915,42 @@ public function addFormatsAndUrl(Media $media)
             $media->setUrl($versionData[$media->getVersion()]['url']);
         }
 
+        // Get preview image and overwrite thumbnails if set
+        /** @var \Sulu\Bundle\MediaBundle\Entity\Media $previewImage */
+        $previewImage = $media->getEntity()->getPreviewImage();
+
+        if ($previewImage !== null) {
+            /** @var FileVersion $latestVersion */
+            $latestVersion = null;
+
+            /** @var File $file */
+            foreach ($previewImage->getFiles() as $file) {
```

Comment:
There should be a version variable on one of the entities containing the latest version, so you don't have to loop over all the versions.